plugin-enclave-gradle/src/main/kotlin/com/r3/conclave/plugin/enclave/gradle/gramine/GenerateGramineEnclaveMetadata.kt

```diff
@@ -0,0 +1,117 @@
+package com.r3.conclave.plugin.enclave.gradle.gramine
+
+import com.r3.conclave.plugin.enclave.gradle.BuildType
+import com.r3.conclave.plugin.enclave.gradle.ConclaveTask
+import com.r3.conclave.utilities.internal.digest
+import org.bouncycastle.openssl.PEMKeyPair
+import org.bouncycastle.openssl.PEMParser
+import org.bouncycastle.openssl.jcajce.JcaPEMKeyConverter
+import org.gradle.api.file.RegularFileProperty
+import org.gradle.api.model.ObjectFactory
+import org.gradle.api.provider.Property
+import org.gradle.api.tasks.Input
+import org.gradle.api.tasks.InputFile
+import org.gradle.api.tasks.Optional
+import org.gradle.api.tasks.OutputFile
+import java.io.File
+import java.security.interfaces.RSAPublicKey
+import java.util.*
+import javax.inject.Inject
+
+/**
+ * In native mode, enclave metadata is bundled into the enclave .so file by the Intel SGX SDK. The Intel SDK then uses
+ * this data when loading the enclave. This allows our code to be mostly agnostic of the enclave mode. With Gramine
+ * though, there is no concept of a "simulation" mode, so we have to implement some of this logic ourselves.
+ * This task produces a properties file which serves as our mechanism for passing build-time information for the host to
+ * use at runtime.
+ * It should be noted that this file is distinct from the enclave properties file in that it is available to the host
+ * *before* the enclave is loaded.
+ *
+ * TODO: Investigate using environment variables in the Manifest to pass this metadata instead.
+ */
+open class GenerateGramineEnclaveMetadata @Inject constructor(objects: ObjectFactory) : ConclaveTask() {
+    companion object {
+        const val METADATA_FILE_NAME = "enclave-metadata.properties"
+    }
+
+    @get:Input
+    val buildType: Property<BuildType> = objects.property(BuildType::class.java)
```

Comment:
Inject this in. The way the plugin is designed is such each mode operates independently of each other, meaning you can create all four modes of the enclave without overwriting each other. Which all means this isn't an input property to tasks (you can see that if the other tasks).